WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift

```diff
@@ -92,16 +92,18 @@ private extension CardReaderSettingsConnectedViewController {
     }
 
     func configureTable() {
+        tableView.separatorStyle = .none
+        tableView.backgroundColor = .systemBackground
         tableView.rowHeight = UITableView.automaticDimension
+        tableView.estimatedSectionFooterHeight = 0
+        tableView.estimatedSectionHeaderHeight = 0
         tableView.dataSource = self
         tableView.delegate = self
         tableView.reloadData()
     }
 
     func configureUpdateView() {
-        guard let viewModel = viewModel else {
-            return
-        }
+        let viewModel = viewModel // capture for closures
```

Comment:
Yes, it's just that this seemed easier to read for me, since we'd have to do the weak/strong dance in two closures, which are otherwise both one-liners.

We could pass in [weak self] to each, then convert `self` to strong here, that would be fine too, and more common which does count for something.

As a third option, we could even pass the view model's functions directly, rather than using closures, like this:

```
    func configureUpdateView() {
        if let error = viewModel.readerUpdateError {
            if case [...] {
                [...]
            } else {
                settingsAlerts.updatingFailed(
                    from: self,
                    tryAgain: viewModel.startCardReaderUpdate,
                    close: viewModel.dismissReaderUpdateError
                )
            }
        [...]
    }
```

On balance, I like what we have already, because it scans a little more clearly as "these are functions for buttons" while avoiding capturing self and all the clutter that comes with it, which would overwhelm these one-line closures.

Reply:
Thanks for the explanation! 🙇